mountpoint-s3/src/prefetch/part_queue.rs

```diff
@@ -73,11 +84,15 @@ impl<E: std::error::Error + Send + Sync> PartQueue<E> {
             Ok(part)
         }
     }
+}
 
+impl<E: std::error::Error + Send + Sync> PartQueueProducer<E> {
     /// Push a new [Part] onto the back of the queue
     pub fn push(&self, part: Result<Part, E>) {
         // Unbounded channel will never actually block
-        self.sender.send_blocking(part).unwrap();
+        if self.sender.send_blocking(part).is_err() {
+            trace!("closed channel");
+        }
```

Comment:
i think we don't expect this to ever happen, but it's not fatal, so maybe `warn!` instead. also style preference/nit, not putting side effects in conditions
```suggestion
        let result = self.sender.send_blocking(part);
        if result.is_err() {
            warn!("closed channel");
        }
```